types/underscore/underscore-tests.ts

```diff
@@ -2443,6 +2451,45 @@ undefinedIdentityIterateeResult; // $ExpectType StringRecord
     extractChainTypes(_.chain(simpleString).without(simpleString[0], simpleString[1])); // $ExpectType ChainType<string[], string>
 }
 
+// union
+{
+    // lists
+    _.union(...recordListArray); // $ExpectType StringRecord[]
+    _(stringRecordList).union(...recordListArray); // $ExpectType StringRecord[]
+    extractChainTypes(_.chain(stringRecordList).union(...recordListArray)); // $ExpectType ChainType<StringRecord[], StringRecord>
+
+    // list and array mix
+    _.union(simpleStringList, simpleStringArray, simpleStringList); // $ExpectType string[]
```

Comment:
I wasted a little time trying to figure out how I could do something like `union<V extends List<any>>(...lists: V[])` and still have this work before deciding it wasn't worth it. I guess maybe at some point it will be possible to do something where `V` is a tuple, but overall it's still just adding an unnecessary level of abstraction that is at best maybe future-proofing the generics just in case `V` is necessary for some reason.

Overall I think I made a mistake trying to get all of these to use `<V extends Collection<any>>` for the sake of consistency with those that needed that for `Iteratee`, and I think as part of the generics cleanup I had mentioned wanting to do I'll reverse course on that where it's not necessary unless you object @jgonggrijp.

Reply:
I'm fine with it either way.